kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/HostapdConfigWriter.java

```diff
@@ -90,16 +94,40 @@ public void visit(NetworkConfiguration config) throws KuraException {
                 .getModifiedNetInterfaceConfigs();
 
         for (NetInterfaceConfig<? extends NetInterfaceAddressConfig> netInterfaceConfig : netInterfaceConfigs) {
-            if (netInterfaceConfig.getType() == NetInterfaceType.WIFI) {
-                if (netInterfaceConfig.getName().startsWith("mon.")) {
-                    continue;
-                }
+
+            // skip monitor interface
+            if (netInterfaceConfig.getName().startsWith("mon.")) {
+                continue;
+            }
+
+            Optional<WifiMode> wifiMode = getWifiMode(netInterfaceConfig);
+
+            if (wifiMode.isPresent() && wifiMode.get() == WifiMode.MASTER
+                    && netInterfaceConfig.getType() == NetInterfaceType.WIFI) {
```

Comment:
Is it better to check if the interface is wifi first?

Reply:
the check is done in getWifiMode

Reply:
Ok. But, so is the check `netInterfaceConfig.getType() == NetInterfaceType.WIFI` useless here?